src/main/java/gregtech/common/covers/CoverPump.java

```diff
@@ -134,10 +135,8 @@ public void update() {
     }
 
     protected int doTransferFluids(int transferLimit) {
-        PooledMutableBlockPos blockPos = PooledMutableBlockPos.retain();
-        blockPos.setPos(coverHolder.getPos()).move(attachedSide);
-        TileEntity tileEntity = coverHolder.getWorld().getTileEntity(blockPos);
-        blockPos.release();
+        BlockPos pos = coverHolder.getPos().offset(attachedSide);
+        TileEntity tileEntity = coverHolder.getWorld().getTileEntity(pos);
```

Comment:
You made an adjacent tile entity cache checker thingy before didn't you dan?

Reply:
It never made it to master branch, I might start looking into it again but it isn't ready yet